tensorboard/webapp/metrics/store/metrics_selectors.ts

```diff
@@ -628,3 +615,30 @@ export const getMetricsCardTimeSelection = createSelector(
     );
   }
 );
+
+export const getSingleSelectionHeaders = createSelector(
+  selectMetricsState,
+  (state: MetricsState): ColumnHeader[] => {
+    return state.singleSelectionHeaders;
+  }
+);
+
+export const getRangeSelectionHeaders = createSelector(
+  selectMetricsState,
+  (state: MetricsState): ColumnHeader[] => {
+    return state.rangeSelectionHeaders;
+  }
+);
+
+export const getColumnHeadersForCard = memoize((cardId: string) => {
+  return createSelector(
+    (state) => state,
+    getSingleSelectionHeaders,
+    getRangeSelectionHeaders,
+    (state, singleSelectionHeaders, rangeSelectionHeaders) => {
+      return getMetricsCardRangeSelectionEnabled(state, cardId)
```

Comment:
So this cannot be input to the selector because it is a selector with a prop?

Reply:
Yeah exactly.